File: lagom/examples/bc1-to-bc2/bc1-to-bc2-impl/src/main/scala/com/apuex/sales/mapping/bc1ToBc2/Bc1ToBc2MappingServiceImpl.scala
package com.apuex.sales.mapping.bc1ToBc2

import com.lightbend.lagom.scaladsl.api.ServiceCall

import scala.concurrent.Future

class Bc1ToBc2MappingServiceImpl extends Bc1ToBc2MappingService {
  /**
    * Subscribe from event stream with offset.
    *
    * @see [[akka.persistence.query.Offset]]
    * @see [[akka.persistence.query.TimeBasedUUID]]
    * @param offset timed-uuid specifies start position
    * @return
    */
  def events(offset: Option[String]) = ServiceCall { inputStream =>
    println(s"offset = ${offset}")
    Future.successful(inputStream.map(x => x))
  }

}

File: lagom/examples/bc1-to-bc2/bc1-to-bc2-impl/src/main/scala/com/apuex/sales/mapping/bc1ToBc2/Bc1ToBc2MappingServiceImpl.java
package com.apuex.sales.mapping.bc1ToBc2;


import akka.NotUsed;
import akka.stream.javadsl.Source;
import com.lightbend.lagom.javadsl.api.ServiceCall;

import javax.inject.Inject;
import java.util.Date;
import java.util.Optional;

import static java.util.concurrent.CompletableFuture.completedFuture;

class Bc1ToBc2MappingServiceImpl implements Bc1ToBc2MappingService {
  @Inject
  private MappingConfig config;
  /**
   * Subscribe from event stream with offset.
   *
   * @param offset timed-uuid or sequence specifies start position
   * @return
   */
  @Override
  public ServiceCall<Source<String, NotUsed>, Source<String, NotUsed>> events(Optional<String> offset) {
    System.out.println(offset);
    return inputs -> completedFuture(
        inputs.map(x -> String.format("[%s] - %s", new Date(),x))
    );
  }

}
